xbmc/platform/posix/filesystem/SMBDirectory.cpp

```diff
@@ -330,3 +344,67 @@ bool CSMBDirectory::Exists(const CURL& url2)
   return S_ISDIR(info.st_mode);
 }
 
+bool CSMBDirectory::GetServerList(CFileItemList &items)
+{
+  char ip[INET6_ADDRSTRLEN];
+  char line[200];
+
+  FILE* fp = popen("nmblookup '*'", "r");
+  
+  if (fp)
+  {
+    while (fgets(line, sizeof line, fp))
+    {
+      if (sscanf(line, "%99s *<00>\n", ip))
+      {
+        if (inet_addr(ip) != INADDR_NONE)
+        {
+          CFileItemPtr pItem(new CFileItem(ip));
```

Comment:
please use shared pointer for new code:
`const CFileItemPtr pItem = std::make_shared<CFileItem>(ip);`

Reply:
well `const` could be wrong here, sorry my bad